bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java

```diff
@@ -335,16 +373,49 @@ private void processConfigFile(File configFile) throws IOException, FileNotFound
                 logger.warn("The file {} subsequently defines the exclusive PID '{}'.", configFile.getAbsolutePath(),
                         exclusivePID);
             }
+
             pid = exclusivePID;
+
+            if (exclusivePID.contains(SERVICE_CONTEXT_MARKER)) {
+                // split pid and context
+                pid = exclusivePID.split(SERVICE_CONTEXT_MARKER)[0];
+                context = exclusivePID.split(SERVICE_CONTEXT_MARKER)[1];
+            }
+
             lines = lines.subList(1, lines.size());
-            exclusivePIDMap.setProcessedPID(pid, configFile.getAbsolutePath());
+            exclusivePIDMap.setProcessedPID(exclusivePID, configFile.getAbsolutePath());

```

Comment:
Its the right thing to add the full PID including the context here. Although, in line 421 (not part of this PR) the exclusivePID map is checked, whether the given PID is existent there: This should prevent the ConfigDispatcher from overwriting already parsed exclusive configs!
Currently this is possible (there is also a test for this).
See line 409 - 411 for further comments.

Reply:
Are you sure that the current implementation skips the content of the second file?

In https://github.com/eclipse/smarthome/blob/master/bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java#L351

It creates an empty `Properties` dictionary and fills it with the content of the second file and then updates the configuration with it.

That's why in this PR I let the duplicate file overwrite the configuration...

Reply:
Yes, I´m sorry! See here: https://github.com/eclipse/smarthome/pull/4685/files#diff-f38a1a84005ec66f7434ebe672d47578R372 for the duplicate check. But it only gives a warning to the log. So overwriting is okay.